### ISSUE TITLE
`unmanaged-cluster` package version resolution should use selection library consistent with kapp

### ISSUE THREAD

**Issue by jpmcb** [vmware-tanzu/community-edition#3912]
## Feature Request

Today, when `unmanaged-cluster` selects a package version (and non was given), it will attempt to select the "latest" semantic versioned package. We are using semver code from Go to accomplish this.

https://github.com/vmware-tanzu/community-edition/blob/bb813fd14b9b79033f035d81f22d2fd570995eb3/cli/cmd/plugin/unmanaged-cluster/tanzu/tanzu.go#L1114-L1118

But this may be inconsistent with versions that kapp controller selects since it uses a different version / semver library. The differences are likely small, but could be noticeable for users doing different package install methods. For example, build metadata may sort differently and create inconsistencies between which packages get selected. 

**_TLDR:_**

In order to have consistency between what version of a package our client selects and what version of a package kapp may possible select, we should use the same version library 

## Additional context

Library used in carvel to resolve versions: https://github.com/vmware-tanzu/carvel-vendir/tree/develop/pkg/vendir/versions

cc @cppforlife
